### PR TITLE
EL-2968 - Sorting Documentation

### PR DESCRIPTION
**Pull request by ashley-hunter**
JIRA: https://jira.autonomy.com/browse/EL-2968

Adding documentation for sort dropdown and sort direction toggle in Angular section. Deprecated old sections.

Also removed "test" folder in our repository root. This contained an old sample project that hasn't been touched in well over a year - not required anymore as we now have plunkers and a seed project.

### CODE REVIEW

File: docs/app/data/components-page.json [hunk 1]
```diff
@@ -1429,16 +1429,41 @@
                     "version": "AngularJS",
                     "deprecated": true
                 },
+                {
+                    "title": "Sorting",
```
Comment (donna-marijne) [inline docs/app/data/components-page.json]: Could we call this "Toolbar Sorting" to give it some context compared to column sorting? And use the link functionality from the deprecated sections.

File: docs/app/pages/components/components-sections/tables/sorting/sorting.component.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        </span>
+    </div>
+</div>
+
+<hr>
+
+<p>
+    A sort dropdown is commonly used to allow the user to choose how they would like a dataset to be ordered.
+    Additionally a direction toggle may also be used to indicate whether or not the data should in ascending or
+    descending order.
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/tables/sorting/sorting.component.html]: Could you mention that this should only be used where Column Sorting (link) is not applicable, e.g. lists of cards?

File: docs/app/pages/components/components-sections/tables/sorting/snippets/app.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    ];
+
+    selected: string = this.options[0];
+    descending: boolean = true;
+
+    /**
+     * Perform any sorting here 
+     */
+    sort(option: string): void {
+        // store the selected option to displaye in the UI
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/tables/sorting/snippets/app.ts]: Typo "displaye"

**Comment by ashley-hunter** [comment]
Review comments addressed

File: docs/app/pages/components/components-sections/tables/sorting/snippets/app.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ export class AppComponent {
      * Perform any sorting here 
      */
     sort(option: string): void {
-        // store the selected option to displaye in the UI
+        // store the selected option to displayed in the UI
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/tables/sorting/snippets/app.ts]: I think it should either be "to display" or "to be displayed".

**Comment by ashley-hunter** [comment]
Fixed typo

**Pull request merged**